test/evp_extra_test2.c

```diff
@@ -1073,6 +1073,36 @@ static int test_evp_md_ctx_copy(void)
     return ret;
 }
 
+#if !defined OPENSSL_NO_DES && !defined OPENSSL_NO_MD5
+static int test_evp_pbe_alg_add(void)
+{
+    int ret = 0;
+    int cipher_nid = 0, md_nid = 0;
+    EVP_PBE_KEYGEN_EX *keygen_ex = NULL;
+    EVP_PBE_KEYGEN *keygen = NULL;
+
+
+    if (!TEST_true(EVP_PBE_alg_add(NID_pbeWithMD5AndDES_CBC, EVP_des_cbc(), EVP_md5(),
+                    PKCS5_PBE_keyivgen)))
+        goto err;
+
+    if (!TEST_true(EVP_PBE_find_ex(EVP_PBE_TYPE_OUTER, NID_pbeWithMD5AndDES_CBC,
+                         &cipher_nid, &md_nid, &keygen, &keygen_ex)))
+
+        goto err;
+
+    if (!TEST_ptr((void*)keygen))
```

Comment:
Compiler is not liking the cast..

Reply:
Yup.  The solution here is to add a couple of functions `TEST_fn_ptr` and `TEST_fn_ptr_null`, with this signature:

``` C
TEST_fn_ptr(void (*fn)(void));
TEST_fn_ptr_null(void (*fn)(void));
```

You will still have to cast when using them, like this:

``` C
    if (!TEST_fn_ptr((void (*)(void))keygen)
```

Reply:
I'd wrap the cast into the wrapper macro so it doesn't have to be everywhere.  However, the cast to `void *` should be good.  Function pointers are allowed to be cast to `void *` and back without changing their representation (in Posix), so this looks like a zealous compiler getting in our way.

Better might be to drop this and test instead `TEST_true(keygen != NULL)` as it was originally but add a comment about function pointers being difficult.

Reply:
> ... so this looks like a zealous compiler getting in our way.

Those compilers are gcc and clang, and they are zealous *beacuse we ask them to* (that's what `--strict-warnings` does).
This particular thing is really a warning (`-Wpedantic`), but since we turn them all into errors...

Reply:
Casting a function pointer to a `void *` is explicitly permitted by the standard.
No warning should be generated IMO.

I still think we revert to the explicit `!= NULL` check and TEST_true and move along.